src/common/mfu_util.c

```diff
@@ -1105,6 +1110,239 @@ bool mfu_is_lustre(const char* path)
+            fclose(mountfp);
+            fprintf(stderr, "Could not stat: %s (%d)", itemp->mfu_mi_mountpoint, errno);
+            return(1);
+        }
+        itemp->mfu_mi_dev = st.st_dev;
+
+        /*
+         *  Now we can insert the item into the list.
+         */
+        list_add_tail(&itemp->mfu_mi_linkage, mfu_mi_list);
+        bytes = getline(&line, &len, mountfp);
+    }
+
+    fclose(mountfp);
+    return(0);
+}
+
+void mfu_init_filesystem_list(void)
+{
+    if (mfu_mi_list_initialized == true) {
+        MFU_LOG(MFU_LOG_INFO, "The filesystem list is already initialized...");
+        return;
+    }
+
+    MFU_LOG(MFU_LOG_INFO, "Initializing filesystem list...");
+    INIT_LIST_HEAD(&mfu_mi_list);
+
+    int rc = mfu_mount_info_get_filesystem_list(&mfu_mi_list);
+    if (rc != 0) {
+        struct mfu_mount_info* itemp = NULL;
+        struct mfu_mount_info* tmpitemp = NULL;
+        list_for_each_entry_safe(itemp,
+                                 tmpitemp,
+                                 &mfu_mi_list,
+                                 mfu_mi_linkage)
+        {
+            list_del_init(&itemp->mfu_mi_linkage);
+            mfu_mount_info_destroy(itemp);
+            mfu_free(&itemp);
+        }
+        return;
+    }
+
+    mfu_mi_list_initialized = true;
+}
+
+void mfu_destroy_filesystem_list(void)
+{
+    if (mfu_mi_list_initialized == false) {
+        MFU_LOG(MFU_LOG_INFO, "The filesystem list is not initialized...");
+        return;
+    }
+
+    struct mfu_mount_info* itemp = NULL;
+    struct mfu_mount_info* tmpitemp = NULL;
+    list_for_each_entry_safe(itemp,
+                             tmpitemp,
+                             &mfu_mi_list,
+                             mfu_mi_linkage)
+    {
+        list_del_init(&itemp->mfu_mi_linkage);
+        mfu_mount_info_destroy(itemp);
+        mfu_free(&itemp);
+    }
+
+    mfu_mi_list_initialized = false;
+}
+
+bool mfu_is_hpss(const char* path)
+{
+    bool is_hpss = false;
+
+#ifdef HPSS_SUPPORT
+    if (mfu_mi_list_initialized == false) {
+        MFU_LOG(MFU_LOG_INFO, "The filesystem list has not been initialized...");
+        return(false);
+    }
+
+    errno = 0;
+    struct stat st;
+    int rc = stat(path, &st);
+    if (rc == -1) {
+        fprintf(stderr, "Could not stat: %s (%d)", path, errno);
+        return(false);
+    }
+
+    struct mfu_mount_info* itemp = NULL;
+    struct mfu_mount_info* tmpitemp = NULL;
+    list_for_each_entry_safe(itemp,
+                             tmpitemp,
+                             &mfu_mi_list,
+                             mfu_mi_linkage)
+    {
+        MFU_LOG(MFU_LOG_DBG, "MFU_MI_DEV: %d FILE_DEV: %d MFU_MI_FSTYPE: %s",
+            itemp->mfu_mi_dev, st.st_dev, itemp->mfu_mi_fstype
+        );
+
+        if ((itemp->mfu_mi_dev == st.st_dev) &&
+            (itemp->mfu_mi_fstype != NULL) &&
+            (itemp->mfu_mi_fssubtype != NULL) &&
+            (strncmp(itemp->mfu_mi_fstype, "fuse", 25) == 0) &&
```

Comment:
That 25 is sticking out to me like a sore thumb. I really should have made a constant there. Anyhow I think it all looks great to me! Thanks for reviewing it and make it fit more with mpifileutils style.

Reply:
Feel free to update this if you want.  I'm good either way.

Reply:
Okay. I'm good either way as well.